tests/test_base_task.py

```diff
@@ -331,3 +332,84 @@ def test_base_pipeline_task_aggregate_upsert_results_upsert_with_errors_noted(
         },
         "qb_upsert_errors": True,
     }
+
+
+def test_base_task_integrity_check_decorator_adds_check_names_to_registered_checks(
+    pandas_task_with_integrity_checks,
+):
+    assert pandas_task_with_integrity_checks._integrity_checks.issuperset(
+        {
+            "expecting_five_items",
+            "expecting_letter_column",
+        }
+    )
+
+
+def test_base_task_run_integrity_checks_invokes_present_and_skips_absent_checks(
+    pandas_task_with_integrity_checks,
+):
+    """
+    This test simulates another task registering an integrity check that this task does
+    not have.  The task's defined integrity checks are invoked, while this other one is
+    quietly skipped.
+    """
+    pandas_task_with_integrity_checks._integrity_checks.add("i_do_not_have")
```

Comment:
The tests are clear overall but I don't fully understand how you know `i_do_not_have` isn't run in this test. Is it just that it doesn't raise an error?

Reply:
Nice catch; honing in on a subtle detail here.  Relevant to our decorators conversation in a recent Transmog discussion, when tasks register an integrity check it's at the `HRQBTask` **class** level.  

So, when any given task calls `self._integrity_checks`, it will get back a list of ALL integrity check method names, across all tasks.  

This may seem problematic at first, but it's only a name.  So when a task starts to run tests, one of two things happen:
1. it has a method with that name, and it runs its own method
2. it does NOT have a method with that name, quietly skips

This was simulating that another task had registered a check with name `i_do_not_have`, and this test just quietly skips it.  I agree it's not immediately obvious, but I think it really doesn't have any bearing on coding logic: any task is free to define whatever method names it wants for integrity checks.  And, there is no "bleed over" where one task runs another, because it's just by name, not a reference to those other methods.

Reply:
Got it, thank you for clarifying!